File: problems/two_pointers/is_palindrome.ts
export function isPalindrome(s: string) {
  let left = 0;
  let right = s.length - 1;

  while (left <= right) {
    let leftChar = s[left];
    let rightChat = s[right];

    if (leftChar === rightChat) {
      left++;
      right--;
      continue;
    }
    return false;
  }

  return true;
}

File: problems/HackerRank/2521_exercises/linkedList/easy/isPalindrome.c
bool isPalindrome(struct ListNode *head)
{

	struct ListNode *slow = head;
	struct ListNode *fast = head;

	// traverse fast pointer until end of the list to find the middle node
	while (fast != NULL && fast->next != NULL)
	{
		fast = fast->next->next;
		slow = slow->next;
	}

	// reverse list from the slow
	struct ListNode *prev = NULL;
	while (slow != NULL)
	{
		struct ListNode *temp = slow->next;
		slow->next = prev;
		prev = slow;
		slow = temp;
	}

	struct ListNode *left = head;
	struct ListNode *right = prev;

	// compare node values
	while (right)
	{
		if (left->val != right->val)
		{
			return false;
		}
		left = left->next;
		right = right->next;
	}
	return true;
}